### ISSUE TITLE
Change in escaping

### ISSUE THREAD

**Issue by awead** [rsolr/rsolr#101]
::solr_escape which is now replacing the deprecated ::escape has one difference:
```
>   RSolr.escape("foo bar")
=> "foo\\ bar"
>   RSolr.solr_escape("foo bar")
=> "foo bar"
```
Is this significant? Or are we supposed to do to `field:foo AND field:bar` in our queries?

**Comment by awead** [comment]
@ndushay just read over #99 and it seems we'll need to treat spaces on our own as our own local needs dictate. If you have suggestions about the blacklight issue, I can patch that accordingly. Feel free to close this issue once you've replied.

**Comment by ndushay** [comment]
I knew that my fix would be a change in behavior, but I wasn't sure how far-reaching it would be.  It's true that the strings produced are different;  I wonder when the expected Solr behavior changes.  For example, does

>defType=lucene&q=float_field:1.11   

give diff Solr results than

>defType=lucene&q=float_field:1\\.11   

I know that 
>defType=lucene&q=text_field:foo bar

is likely to give diff results than

>defType=lucene&q=text_field:foo\\ bar

but maybe the calling code should be doing this

>defType=lucene&q=text_field:(foo bar)

or this

>defType=lucene&q=text_field:"foo bar"

depending on desired outcomes.


Perhaps documentation could be our friend?  Or perhaps this should NOT deprecate RSolr.escape until an RSolr 2.x release?

Solr escaping is supposed to be only for those characters that have specific meaning to Solr query parsing.  Spaces and periods fall into the category of characters that were formerly incorrectly escaped - they supposedly do not have special meaning to the Solr query parser.  Or, at least, the Solr code I looked at (e.g. SolrJ) didn't escape them.

If we have downstream RSolr consumers that want the larger set of chars escaped, we could keep both methods, or have

solr_escape_minimal
solr_escape_aggressive

or something.

I am fine with any approach that includes a way to get "correct" Solr escaping via RSolr.  Sorry if the patch got merged too quickly without enough discussion.

**Comment by awead** [comment]
@ndushay I'm :sunglasses: with the patch. It's only a deprecation warning anyway! I think it's best if RSolr adhere to what the official escaping practices are as determined in Solr itself, meaning don't escape periods and spaces. Leave it up to the implementing application to decide what to do. I've already done this in ActiveFedora: active_fedora#722 I'll wait for @jcoyne to weigh in on that.

The blacklight issue I refer to @cbeer. The tests fail with `1\.11` versus `1.11` but I don't know the actually fail in Solr as well. I'll need to check on that.

**Comment by cbeer** [comment]
I haven't followed the code paths yet, but we should make sure it's easy for clients to produce parameters like `q=text_field:(foo bar)` or `q=text_field:"foo bar"`.

I suspect with this escaping change, we've probably made that more difficult, but it is probably a common pattern for applications accepting user input.

**Comment by ndushay** [comment]
I agree with @cbeer .   It should also be easy to do   text_field:foo text_field:bar which may be the best capturing of the default intent.
